### ISSUE TITLE
Sort list of properties in alphabetical order

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#9290]
The purpose of documentation is for people to read it.

Some checks have many properties, for example [WhitespaceAround](https://checkstyle.org/config_whitespace.html#WhitespaceAround_Properties). It is hard to find the right one when the list is unsorted.
The list of properties should be sorted.

Maybe there should be one exception for the `tokens` property - it may be the last one for readability.

@romani @rnveach @strkkk @nmancus1 please share your opinion.

**Comment by strkkk** [comment]
Ok for me (with tokens at the bottom).

**Comment by rnveach** [comment]
Sorting for list of tokens on xdoc comes directly from check (minus any that can't be controlled by user).
https://github.com/checkstyle/checkstyle/blob/1e3fb3fae940bf7ea340592f7f0cda3af1b320e3/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java#L498

Instead of sorting the XDoc's tokens, I say we need to sort the tokens defined in the Check's methods (default, required, acceptable) including the Javadoc tokens.

Just like reading the XDoc is hard, the Java code is no different as it lists the exact same tokens, probably more in quantity, and the same order. Once we sort the code, the XDoc will automatically be required to be updated in the same way. If it isn't for whatever reason, then we need to ensure it maintains the same order.

**Comment by nmancus1** [comment]
> The list of properties should be sorted.

I agree, if for no other reason than it would put similar properties together, such as `allows...` and `ignore...`.

> Maybe there should be one exception for the` tokens` property - it may be the last one for readability.

Agree here too, the list of tokens is often fairly long, so this makes sense.  I don't think there are many properties that start with a letter after "t" anyway, so this will probably happen naturally.

> Instead of sorting the XDoc's tokens, I say we need to sort the tokens defined in the Check's methods...

I agree with this point, in fact many are like this already (but many are not). 

So, in order to resolve this issue, we would need to make four changes:

1. Create check instance to enforce alphabetized properties in check documentation, excluding `tokens` property
2. Create check instance to enforce that tokens is the last property
3. Create check instance to enforce alphabetized array initialization for tokens in each check
4. Fix violations from above

Could we use [MatchXPathCheck](https://checkstyle.sourceforge.io/config_coding.html#MatchXpath) to check this?  If so, it would be good to add these as examples to the MatchXPath documentation.

**Comment by rnveach** [comment]
These don't need to be checks. Just add code to XdocsPagesTest and AllChecksTest.

**Comment by romani** [comment]
I am ok with this, please beware of properties from abstract classes like `violateExecutionOnNonTightHtml`
https://checkstyle.org/config_javadoc.html#SummaryJavadoc_Properties they also need to be somewhere on bottom.

**Comment by aryaniiit002** [comment]
so if somebody add code in XdocsPagesTest and AllChecksTest then it's done no matter in what order properties are written in [xdocs](https://github.com/checkstyle/checkstyle/tree/master/src/xdocs) ?

thumbs up emoji will be considered as `yes`.

**Comment by rnveach** [comment]
If you add code to the tests, the test code added must enforce the requested changes in the xdocs. Since most xdocs don't follow this order, that means you will need to make changes to the xdocs. Your new test code should guide you on what xdoc changes to make, not the other way around.

**Comment by romani** [comment]
already sorted https://checkstyle.org/checks/javadoc/javadocmethod.html
